Update YamlCreate.ps1 (Listen For Keypress)

vedantmgoyal9 opened the pull request:
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----
From #21213

@OfficialEsco Sorry for that mistake. Here is a working fix.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/21258)

vedantmgoyal9 committed:
Update YamlCreate.ps1

OfficialEsco commented:
Personally i don't see the reason to use `Yaml.Create` in those PRs, its not matching wingetcreate and none else is using it but you

ImJoakim commented:
> Personally i don't see the reason to use `Yaml.Create` in those PRs, its not matching wingetcreate and none else is using it but you

I agree with this. In any case, you can make that change on your local machine and use it yourself.

ItzLevvie reviewed (changes requested):
https://github.com/microsoft/winget-pkgs/pull/21258#issuecomment-881859853

vedantmgoyal9 committed:
Update YamlCreate.ps1

vedantmgoyal2009 commented:
Made some more changes from my side.

vedantmgoyal9 committed:
Revert Commit a168868

Undo Commit Message Modification

OfficialEsco reviewed (changes requested):
```
[Optional] Enter the application product code. Looks like {CF8E6E00-9C03-4440-81C0-21FACB921A6B}
ProductCode found from installer:
Can be found with get-wmiobject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name -AutoSize
ProductCode:
Scope
[Optional] Enter the Installer Scope.
[M] Machine  [U] User  [N] No idea (default is 'N'): uUpgradeBehavior
[Optional] Enter the UpgradeBehavior.
[I] install  [U] uninstallPrevious (default is 'I'): i
Another Installer
[Optional] Do you want to create another installer?
[Y] Yes  [N] No  [?] Help (default is "N"):
```

* Needs more spacing
* UpgradeBehaviour ends up on the same line as the Scope input field
* UpgradeBehaviour got ignored
![image](https://user-images.githubusercontent.com/15158490/126041629-ad01d5ec-88d3-45bd-8278-6c263973a54c.png)

vedantmgoyal2009 commented:
Needs more spacing - Fixed
UpgradeBehaviour ends up on the same line as the Scope input field - Fixed
UpgradeBehaviour got ignored  - Fixed

Working on few more changes, will commit the new changes soon. 🙂

vedantmgoyal9 committed:
Made more changes

Fixed the following things-
- Needs more spacing
- UpgradeBehaviour ends up on the same line as the Scope input field
- UpgradeBehaviour got ignored

Added Listen for keypress during Scope, UpgradeBehavior, AnotherInstaller

OfficialEsco commented:
Not sure how you managed to do this but when i entered the InstallModes it looped me
```
[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:

[Optional] Enter any Protocols the application provides a handler for. For example: http, https (Max 16)
Protocols:

[Optional] Enter any Commands or aliases to run the application. For example: msedge (Max 16)
Commands:

[Optional] List of additional non-zero installer success exit codes other than known default values by winget (Max 16)
InstallerSuccessCodes:

[Optional] List of supported installer modes. Options: interactive, silent, silentWithProgress
InstallModes: interactive, silent

[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:

[Optional] Enter any Protocols the application provides a handler for. For example: http, https (Max 16)
Protocols:

[Optional] Enter any Commands or aliases to run the application. For example: msedge (Max 16)
Commands:

[Optional] List of additional non-zero installer success exit codes other than known default values by winget (Max 16)
InstallerSuccessCodes:

[Optional] List of supported installer modes. Options: interactive, silent, silentWithProgress
Old Variable: interactive, silent
InstallModes:

[Optional] Enter the full publisher name. For example: Microsoft Corporation
Old Variable: Wiris
Publisher:
```

vedantmgoyal2009 commented:
Thanks for your feedback. I will into it and push a fix soon.

vedantmgoyal9 committed:
Fixed Looping after `InstallModes`

vedantmgoyal9 committed:
Listen For Keypress at `Save to disk`

Add Listen For Keypress at "Save to disk?" Prompt.

vedantmgoyal9 committed:
Listen for keypress at `Sandbox Test` and `Submit PR?`

vedantmgoyal2009 commented:
I have fixed the looping issue. If there is any issue now, please tell me. I have tried my best to reduce the number of times a user has to press enter↵ while creating a manifest. I have implemented Trenly's logic from #21143.

OfficialEsco commented:
@quhxl can you test it?

quhxl commented:
Before:
![before](https://user-images.githubusercontent.com/69106310/126047417-94d90e33-5dd4-4372-8c08-e8428ed30fdd.png)
After:
![after](https://user-images.githubusercontent.com/69106310/126047419-1c48a179-bdce-4dfb-ad8f-51da0b7e05e4.png)

@OfficialEsco Seems to be fixed. @vedantmgoyal2009 But I also got looped at `FileExtensions` to `InstallMode`.

vedantmgoyal2009 commented:
Thanks, I will look into the looping issue again.

OfficialEsco commented on Tools/YamlCreate.ps1:
```diff
@@ -373,19 +397,24 @@ Function Read-WinGet-InstallerValues {
-    if ($AnotherInstaller -eq '0') {
-        Read-WinGet-InstallerValues
+    if ($keyInfo.Key -eq 'Y') {
+        Write-Host; Read-WinGet-InstallerManifest
```
```suggestion
        Write-Host; Read-WinGet-InstallerValues
```
```
Another Installer
[Optional] Do you want to create another installer?
[Y] Yes  [N] No (default is 'N'): y


[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:
```
This is inverted, If Yes loop back

vedantmgoyal2009 replied:
You mean the issue is fixed now?

OfficialEsco replied:
This suggestion is still Inverted

My other issue seems to be solved, not sure where quhxl got looped since he didn't provide log

OfficialEsco replied:
```
Another Installer
[Optional] Do you want to create another installer?
[Y] Yes  [N] No (default is 'N'): n

[Optional] Enter any File Extensions the application could support. For example: html, htm, url (Max 256)
FileExtensions:
```
Nvm

OfficialEsco replied:
the last 2 commits did indeed solve the issue

vedantmgoyal2009 replied:
I have committed changes.

vedantmgoyal9 committed:
Fix Looping using switch and default

vedantmgoyal2009 commented:
I have fixed it using another way. Now, it will work in any case.

vedantmgoyal9 committed:
Fixed function calling at `AnotherInstaller`

OfficialEsco commented:
Seems to be stable for me now

This part needs a bit more space
```
Save to disk?
Do you want to save the files to the Temp folder?
[Y] Yes  [N] No  [M] Manually Enter SHA256 (default is 'N'):
Downloading URL. This will take a while...
Time taken: 3 second(s)
```

This part got one space too many 😁
```
[Optional] Enter the application product code. Looks like {CF8E6E00-9C03-4440-81C0-21FACB921A6B}
ProductCode found from installer:
Can be found with get-wmiobject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name -AutoSize
ProductCode:


Scope
[Optional] Enter the Installer Scope.
[M] Machine  [U] User  [N] No idea (default is 'N'): u
```

And this part needs a bit more space
```
Yaml file created: C:\Users\odin9\GitHub\winget-pkgs\manifests\e\Eugeny\Tabby\1.0.147\Eugeny.Tabby.locale.en-US.yaml
Manifest validation succeeded.
Submit PR?
Do you want to submit your PR now?
Switched to a new branch 'Eugeny.Tabby-1.0.147'
```

vedantmgoyal9 committed:
Spacing added

vedantmgoyal2009 commented:
I think now, it is working perfectly.

vedantmgoyal9 committed:
Metadata: Zint.Zint version 2.4.2.0

vedantmgoyal9 committed:
Revert "Metadata: Zint.Zint version 2.4.2.0"

This reverts commit d05fae43b781ca876f9271a167f66f260d688640.

vedantmgoyal2009 commented:
Sorry, by mistake, committed manifests to this branch.

denelon commented:
@OfficialEsco I'll wait to review/approve/merge until after you validate and approve.

OfficialEsco reviewed (approved):
Seems to be working correctly in 3 Updates and 1 New Manifest, i have not tested the Sandbox prompt tho
